go/vt/vtgateproxy/discovery.go

```diff
@@ -108,7 +159,7 @@ var (
 	targetCount = stats.NewGaugesWithSingleLabel("JsonDiscoveryTargetCount", "Count of hosts returned from discovery by pool type", "pool")
 )
 
-func RegisterJSONGateResolver(
+func RegisterGateResolver(
```

Comment:
High-level -- I think it might make more sense to clone this implementation and fork from there rather than modify in-place.

Specifically for rollout purposes, I think it's better to have both json and rotor discovery available in a given dbproxy build where we can pick one or the other at runtime based on config (or mayyyybe even running both in parallel for comparisons?).